pyEpiabm/pyEpiabm/sweep/intervention_sweep.py

```diff
@@ -36,22 +36,36 @@ def __init__(self):
         """
         # Implemented interventions and their activity status
         self.intervention_active_status = {}
-        self.intervention_params = Parameters.instance().intervention_params
+        self.intervention_params = Parameters.instance().\
+            intervention_params.copy()
+        self.intervention_dict = {'case_isolation': CaseIsolation,
+                                  'place_closure': PlaceClosure,
+                                  'household_quarantine': HouseholdQuarantine,
+                                  'social_distancing': SocialDistancing,
+                                  'disease_testing': DiseaseTesting,
+                                  'vaccine_params': Vaccination,
+                                  'travel_isolation': TravelIsolation}
 
     def bind_population(self, population):
         self._population = population
-        intervention_dict = {'case_isolation': CaseIsolation,
-                             'place_closure': PlaceClosure,
-                             'household_quarantine': HouseholdQuarantine,
-                             'social_distancing': SocialDistancing,
-                             'disease_testing': DiseaseTesting,
-                             'vaccine_params': Vaccination,
-                             'travel_isolation': TravelIsolation}
-
         for intervention in self.intervention_params.keys():
-            params = self.intervention_params[intervention]
-            self.intervention_active_status[(intervention_dict[intervention](
-                population=self._population, **params))] = False
+            if isinstance(self.intervention_params[intervention],
+                          list):
+                for index in range(len(self.intervention_params[
+                        intervention])):
+                    params = self.intervention_params[intervention][index]
+                    intervention_init = self.intervention_dict[intervention](
+                            population=self._population, **params)
+                    intervention_init.occurrence_index = index
+                    self.intervention_active_status[intervention_init] = False
+                    if index == 0:
```

Comment:
This could do with a comment why you are modifying the original parameters object, and for one index only. Once this intervention occurs, do you update the parameters object to store the details of the next intervention in the list. Can the interventions be concurrent and do you check for this if not?

Reply:
I've added the a comment in the code. Each type of interventions are assumed to be in a sequence, not concurrent and when such intervention is activated, the parameter will be updated. **Concurrent** could handle by the input json file. For instance: from day1-day10 only close School, from day 5-15 only close outdoor. Then parameter will be given as: start_date = 1, duration = 4, close_type = [school]; start_date = 5, duration = 5, close_type = [school, outdoor]; start_date = 10, duration = 5, close_type = [outdoor].

Reply:
> I've added the a comment in the code. Each type of interventions are assumed to be in a sequence, not concurrent and when such intervention is activated, the parameter will be updated. **Concurrent** could handle by the input json file. For instance: from day1-day10 only close School, from day 5-15 only close outdoor. Then parameter will be given as: start_date = 1, duration = 4, close_type = [school]; start_date = 5, duration = 5, close_type = [school, outdoor]; start_date = 10, duration = 5, close_type = [outdoor].

I'm not quite sure what you mean that "Concurrent could handle by the input json file". I think if we are okay with concurrent inputs in the json then we need a test to check that they work as expected. Otherwise we need to raise an error somewhere (or even just flag a warning saying the second one is ignored or similar). I am happy with you allowing concurrent inputs, but I do think this is a tricky case for the code, and so I think we need to test that concurrent interventions
 can be active simultaneously, activated and deactivated separately (at the correct times), and have a summed effect when both are active.

Reply:
From current code, the intervention will last for 15 days, but parameters would be changed from day 5 (intervention_para1 from day1 to day4, intervention_para2 from day5 to day 15). Now I raised such error message by checking whether the previous same_category_intervention'd end date is later than the current intervention's start date.